modules/ddb_cover_upload/includes/ddb_cover_upload.uploaded_list.inc

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/**
+ * @file
+ * List display of uploaded covers from this library.
+ */
+
+use CoverServiceUpload\Api\MaterialApi;
+use GuzzleHttp\Client;
+
+/**
+ * Display list of covers uploaded by the library.
+ *
+ * @return array
+ *   A themed list.
+ *
+ * @throws \CoverServiceUpload\ApiException
+ */
+function ddb_cover_upload_list() {
+  $rows = array();
+
+  $materials = _ddb_cover_upload_get_materials();
+  foreach ($materials as $material) {
+    $cover = $material->getCover();
+    $image = array(
+      '#theme' => 'image',
+      '#path' => $cover->getImageUrl(),
+      '#width' => '120px',
+    );
+
+    /** @var \Ting\TingObject $object */
+    $object = ding_entity_load($material->getIsIdentifier(), 'ting_object');
+    $title = t('Not found');
+    $link = $material->getIsIdentifier();
+    if (FALSE !== $object) {
+      $uri = ding_entity_uri('ding_entity', $object);
+      $title = $object->getTitle() . ' (' . $object->getType() . ')';
+      $link = l($material->getIsIdentifier(), $uri['path'], array('html' => TRUE));
+    }
+
+    $rows[] = [
+      $material->getId(),
+      render($image),
+      $title,
+      $link,
+      l(t('Delete'), 'admin/config/cover_upload/upload/list/delete/' . $material->getId(), array('html' => TRUE)),
+    ];
+  }
+
+  drupal_set_title(t('List uploaded covers'));
+  $header = array(
+    t('Id'),
+    t('Image'),
+    t('Title'),
+    t('Material'),
+    t('Actions'),
+  );
+
+  return array(
+    '#markup' => theme('table', array(
+      'header' => $header,
+      'rows' => $rows,
+    ))
+  );
+}
+
+/**
+ * Menu callback for deleting an uploaded cover.
+ */
+function ddb_cover_upload_delete($form, &$form_state, $id) {
+  $form['material_id'] = array(
+    '#type' => 'value',
+    '#value' => $id,
+  );
+
+  return confirm_form(
+    $form,
+    t('Are you sure you want to delete this item?'),
+    'admin/config/cover_upload/upload/list',
+    t('This action cannot be undone.'),
+    t('Delete'),
+    t('Cancel')
+  );
+}
+
+/**
+ * Confirm form submit handler for delete above.
+ */
+function ddb_cover_upload_delete_submit($form, &$form_state) {
+  if (isset($form_state['values']['material_id'])) {
+    _ddb_cover_upload_delete_material($form_state['values']['material_id']);
```

Comment:
Have you considered writing to `watchdog` that the cover has been deleted?

Reply:
No, as there is a lot of logging at the service. So added watchdog statement.